clustering/clustering.go

```diff
@@ -110,7 +110,8 @@ func geocluster(clusters []*Cluster, threshold float64) []*Cluster {
 		clusters,
 		// Default intialization, TODO: better bucketing/prefiltering will greatly increase performance.
 		// can use the bound above to help with this.
-		initClusterDistances(clusters, CentroidSquaredDistance{}, scaledThreshold),
+		// initClusterDistances(clusters, CentroidSquaredDistance{}, scaledThreshold),
+		initGeoClusterDistances(bound, clusters, CentroidSquaredDistance{}, scaledThreshold),
```

Comment:
you can toggle these lines and run `go test -bench ClusterGeo` so see the performance differences.